### PR TITLE
feat(appeals): update S78 timetable (A2-661)

### PR DESCRIPTION
**Pull request by fabfab-kc**
Updates the S78 timetable

- Refactors fields in database, so they are updateable individually
- Maps the new field to API and WEB update model
- Adds mappers for specific dates

## Issue ticket number and link

[A2-661](https://pins-ds.atlassian.net/browse/A2-661)

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

[A2-661]: https://pins-ds.atlassian.net/browse/A2-661?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: appeals/web/src/server/lib/mappers/appeal/has.js [hunk 1]
```diff
@@ -22,7 +22,6 @@ import { mapCostsLpaCorrespondence } from './submappers/costs-lpa-correspondence
 import { mapCostsLpaWithdrawal } from './submappers/costs-lpa-withdrawal.mapper.js';
 import { mapCrossTeamCorrespondence } from './submappers/cross-team-correspondence.mapper.js';
 import { mapDecision } from './submappers/decision.mapper.js';
-import { mapFinalCommentReviewDueDate } from './submappers/final-comment-review-due-date.mapper.js';
```
Comment (Si3rr4wow) [inline appeals/web/src/server/lib/mappers/appeal/has.js:25]: Are the mappers you removed from here redundant now?

File: appeals/web/src/server/lib/mappers/appeal/submappers/s106-obligation-due-date.mapper.js [hunk 2]
```diff
@@ -10,14 +10,14 @@ export const mapS106ObligationDue = ({ appealDetails, currentRoute }) => ({
 			},
 			value: {
 				// @ts-ignore
-				html: dateISOStringToDisplayDate(appealDetails.appealTimetable?.s106ObligationDue) || ''
+				html: dateISOStringToDisplayDate(appealDetails.appealTimetable?.s106ObligationDueDate) || ''
```
Comment (Si3rr4wow) [inline appeals/web/src/server/lib/mappers/appeal/submappers/s106-obligation-due-date.mapper.js:13]: are we able to remove the optional chaining now?

File: appeals/web/src/server/lib/mappers/appeal/has.js (hunk 1 above)
Comment (fabfab-kc) [reply appeals/web/src/server/lib/mappers/appeal/has.js:25]: I've realigned the timetable so we have these deadlines:

-       lpaQuestionnaireDueDate?: string | null;
- 	ipCommentsDueDate?: string | null;
- 	appellantStatementDueDate?: string | null;
- 	lpaStatementDueDate?: string | null;
- 	appellantFinalCommentsDueDate?: string | null;
- 	lpaFinalCommentsDueDate?: string | null;
- 	s106ObligationDueDate?: string | null;
- 	issueDeterminationDate?: string | null;

and realigned the mappers accordingly... I noticed that we only showing the lpaStatementDueDate, not the appellantStatement one...

File: appeals/web/src/server/lib/mappers/appeal/submappers/s106-obligation-due-date.mapper.js (hunk 2 above)
Comment (fabfab-kc) [reply appeals/web/src/server/lib/mappers/appeal/submappers/s106-obligation-due-date.mapper.js:13]: mmm, what do you mean? It's just the field renamed, this wasn't in the DB

Comment (Si3rr4wow) [reply appeals/web/src/server/lib/mappers/appeal/submappers/s106-obligation-due-date.mapper.js:13]: removing the `?`

Comment (fabfab-kc) [reply appeals/web/src/server/lib/mappers/appeal/submappers/s106-obligation-due-date.mapper.js:13]: I see, yes you're right

Comment (fabfab-kc) [reply appeals/web/src/server/lib/mappers/appeal/submappers/s106-obligation-due-date.mapper.js:13]: ignore, timetable is null until the case is started...

File: appeals/web/src/server/lib/mappers/appeal/submappers/appellant-final-comment-due-date.mapper.js [hunk 3]
```diff
@@ -11,15 +11,16 @@ export const mapAppellantFinalCommentDueDate = ({ appealDetails, currentRoute })
 			value: {
 				html:
 					// @ts-ignore
```
Comment (james-hill-kc) [inline appeals/web/src/server/lib/mappers/appeal/submappers/appellant-final-comment-due-date.mapper.js:13]: I know this wasn't added in this PR, but it looks like this `ts-ignore` could be removed if `appellantFinalCommentDueDate` was added to the `WebAppealTimetable` type def in `appeals/web/src/server/appeals/appeal-details/appeal-details.types.d.ts`

File: appeals/web/src/server/lib/mappers/appeal/submappers/lpa-final-comment-due-date.mapper.js [hunk 4]
```diff
@@ -11,8 +11,8 @@ export const mapLpaFinalCommentDueDate = ({
 		id: 'lpa-final-comment-due-date',
 		text: 'LPA final comments due',
 		// @ts-ignore
```
Comment (james-hill-kc) [inline appeals/web/src/server/lib/mappers/appeal/submappers/lpa-final-comment-due-date.mapper.js:13]: The same type fix as mentioned elsewhere applies here (just need to add `lpaFinalCommentDueDate` to `WebAppealTimetable`)

**Review by james-hill-kc — COMMENTED** [review]
Looks good to me! I've added some comments but they all relate to removing `ts-ignores`. They weren't added in this PR but it's probably worth fixing them as part of it.

**Review by james-hill-kc — APPROVED** [review]
LGTM!

**Pull request merged**